swiftide/tests/lancedb.rs

```diff
@@ -87,3 +92,125 @@ async fn test_lancedb() {
         codefile.to_str().unwrap()
     );
 }
+
+#[test_log::test(tokio::test)]
+async fn test_lancedb_retrieve_dynamic_search() {
+    // Setup temporary directory and file for testing
+    let tempdir = TempDir::new().unwrap();
+    let codefile = tempdir.child("main.rs");
+    let code = "fn main() { println!(\"Hello, World!\"); }";
+    std::fs::write(&codefile, code).unwrap();
+
+    // Setup mock servers to simulate API responses
+    let mock_server = MockServer::start().await;
+    mock_chat_completions(&mock_server).await;
+
+    let openai_client = openai_client(&mock_server.uri(), "text-embedding-3-small", "gpt-4o");
+
+    let fastembed = FastEmbed::try_default().unwrap();
+
+    let lancedb = LanceDB::builder()
+        .uri(tempdir.child("lancedb").to_str().unwrap())
+        .vector_size(384)
+        .with_vector(EmbeddedField::Combined)
+        .with_metadata(METADATA_QA_CODE_NAME)
+        .with_metadata("filter")
+        .with_metadata("path")
+        .table_name("swiftide_test")
+        .build()
+        .unwrap();
+
+    Pipeline::from_loader(loaders::FileLoader::new(tempdir.path()).with_extensions(&["rs"]))
+        .then_chunk(ChunkCode::try_for_language("rust").unwrap())
+        .then(MetadataQACode::new(openai_client.clone()))
+        .then(|mut node: indexing::Node| {
+            // Add path to metadata, by default, storage will store all metadata fields
+            node.metadata
+                .insert("path", node.path.display().to_string());
+            node.metadata
+                .insert("filter".to_string(), "true".to_string());
+            Ok(node)
+        })
+        .then_in_batch(transformers::Embed::new(fastembed.clone()).with_batch_size(20))
+        .log_nodes()
+        .then_store_with(lancedb.clone())
+        .run()
+        .await
+        .unwrap();
+
+    let db_connection = lancedb
+        .get_connection()
+        .await
+        .expect("Failed to establish connection to LanceDB pool");
+
+    let vector_table = db_connection
+        .open_table("swiftide_test")
+        .execute()
+        .await
+        .expect("Failed to open vector search table 'swiftide_test'");
+
+    let similarity_config =
+        swift_query_pipeline::search_strategies::SimilaritySingleEmbedding::from_filter(
+            "filter = \"true\"".to_string(),
+        );
+
+    // Create the custom query strategy for vector similarity search
+    let vector_search_strategy =
+        swift_query_pipeline::search_strategies::CustomStrategy::from_query(
+            move |query_node| -> Result<lance_query_builder::VectorQuery> {
```

Comment:
If that's the case, maybe (thinking very meta), it's possible to yield the strategy here too. And then having the concrete implementation of the strategy (so specific for lance), yield an active table connection.

Honestly, it sounds rather complicated, I don't have an off the bat solution